### ISSUE TITLE
Explosive bubbles

### ISSUE THREAD

**Issue by deferred-impact** [deferred-impact/battledudes-issue-tracker#195]
Medkit renders a bubble. Ammo pack renders a bubble. Trophy system renders a bubble.

Why don't explosions?

Sure, unrealistic. Realism is shit and if you want, i can show you where it belongs, biologically speaking.

Rendering the bubbles matching the explosion range would help people understand it (pretty much nobody currently understands it, let's be honest) and would make the game feel more fair.

Also ZR did it, and it worked really well there.

**Comment by Gregory-Y** [comment]
Good idea, it might fix the inconsistencies that currently plague grenade area damage which I mentioned in #176 . However, I have a concern (which may or may not be valid), do the bubbles appear before the grenades explode or for about a second after they explode? If so, what about rockets and tank shells? Would every rocket have a small circle around it? Because I feel like that would look pretty awkward.

**Comment by deferred-impact** [comment]
I think it's best for the bubble to rapidly flash immediately as the explosion goes off. For like 5-10 frames (~100ms). ZR did something similar, there the bubbles grew in size once the grenade exploded. I don't quite like the growing aspect, but yes it should be quick.

**Comment by deferred-impact** [comment]
Thinking about this some more, perhaps the bubble should be coded such that it takes walls into account, and does not extend past walls which are blocking the damage. Basically the closer it is to the actual gameplay of the explosion, the better.
